src/util/tree.js

```diff
@@ -11,7 +11,11 @@ exports.serialize = (dagNode, callback) => {
   Object.keys(dagNode).forEach((name) => {
     entries.push([name, dagNode[name]])
   })
-  entries.sort((a, b) => a[0] > b[0] ? 1 : -1)
+  entries.sort((a, b) => {
+    const aName = a[0] + (a[1].mode === '40000' ? '/' : '')
+    const bName = b[0] + (b[1].mode === '40000' ? '/' : '')
```

Comment:
I'm not 100% certain if `40000` is the only mode directories can have, but let's call it good enough

Reply:
> I'm not 100% certain if `40000` is the only mode directories can have, but let's call it good enough

Found the [list of supported modes](https://github.com/git/git/blob/ecbdaf0899161c067986e9d9d564586d4b045d62/Documentation/git-fast-import.txt#L555), which specifies 040000 as the only mode for subdirectories.